earthmover/operations/row.py

```diff
@@ -124,3 +124,48 @@ def execute(self, data: 'DataFrame', **kwargs):
                 raise
 
             return data.sort_values(by=self.columns_list, ascending=(not self.descending))
+
+
+class FlattenOperation(Operation):
+    """
+
+    """
+    allowed_configs: Tuple[str] = (
+        'operation', 'repartition', 
+        'flatten_column', 'left_wrapper', 'right_wrapper', 'separator', 'value_column', 'trim_whitespace'
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # Only 'column' or 'columns' can be populated
+        self._flatten_column  = self.error_handler.assert_get_key(self.config, 'flatten_column', dtype=str, required=True)
+        self._left_wrapper = self.error_handler.assert_get_key(self.config, 'left_wrapper', dtype=str, required=False, default="[\"")
+        self._right_wrapper = self.error_handler.assert_get_key(self.config, 'right_wrapper', dtype=str, required=False, default="\"]")
+        self._separator = self.error_handler.assert_get_key(self.config, 'separator', dtype=str, required=False, default=',')
+        self._value_column = self.error_handler.assert_get_key(self.config, 'value_column', dtype=str, required=True)
+        self._trim_whitespace = self.error_handler.assert_get_key(self.config, 'trim_whitespace', dtype=str, required=False, default=" \t\r\n\"")
+
+
+    def execute(self, data: 'DataFrame', **kwargs) -> 'DataFrame':
+        """
+
+        :return:
+        """
+        super().execute(data, **kwargs)
+
+        target_dtypes = data.head(1).rename(columns={self._flatten_column: self._value_column})
```

Comment:
I wonder if there's a way to extract the schema directly without taking a head.

Reply:
updated! and great suggestion